system-addon/bootstrap.js

```diff
@@ -69,22 +84,22 @@ function onPrefChanged(isEnabled) {
 this.install = function install(data, reason) {};
 
 this.startup = function startup(data, reason) {
+  // Only start Activity Stream up when the browser UI is ready
+  Services.obs.addObserver(observe, BROWSER_READY_NOTIFICATION);
+
   // Cache startup data which contains stuff like the version number, etc.
   // so we can use it when we init
   startupData = data;
+  startupReason = reason;
 
   // Listen for changes to the pref that enables Activity Stream
   Preferences.observe(ACTIVITY_STREAM_ENABLED_PREF, onPrefChanged);
```

Comment:
To be extra safe, should we maybe return early in the pref handler if the browser isn't ready yet?